core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -126,7 +127,15 @@ class SparkHadoopUtil extends Logging {
   def getSecretKeyFromUserCredentials(key: String): Array[Byte] = { null }
 
   def loginUserFromKeytab(principalName: String, keytabFilename: String) {
-    UserGroupInformation.loginUserFromKeytab(principalName, keytabFilename)
+    var principal = SecurityUtil.getServerPrincipal(principalName, InetAddress.getLocalHost)
+    UserGroupInformation.loginUserFromKeytab(principal, keytabFilename)
+  }
+
+  def loginUserFromKeytabAndReturnUGI(
+        principalName: String,
+        keytabFilename: String): UserGroupInformation = {
+    var principal = SecurityUtil.getServerPrincipal(principalName, InetAddress.getLocalHost)
```

Comment:
you can't do this for normal operation. A user principal is not the same as a server principal and normally users principals don't have HOSTS.  I don't see how this doesn't break the current functionality.

As I believe @vanzin stated in the jira _HOST is meant for servers, not for applications running on yarn.